setup.py

```diff
@@ -8,133 +8,35 @@
-    except ImportError:
-        exit_with_error(
-            "You need %(pkgname)s %(pkgver)s or greater to run bcolz!"
-            % {'pkgname': pkgname, 'pkgver': pkgver})
-    else:
-        if mod.__version__ < pkgver:
-            exit_with_error(
-                "You need %(pkgname)s %(pkgver)s or greater to run bcolz!"
-                % {'pkgname': pkgname, 'pkgver': pkgver})
-
-    print("* Found %(pkgname)s %(pkgver)s package installed."
-          % {'pkgname': pkgname, 'pkgver': mod.__version__})
-    globals()[pkgname] = mod
-
-
-########### Check versions ##########
-
-# Check for Python
-if sys.version_info[0] == 2:
-    if sys.version_info[1] < 6:
-        exit_with_error("You need Python 2.6 or greater to run bcolz!")
-    if sys.version_info[1] < 7:
-        try:
-            import unittest2
-        except ImportError:
-            print("You need unittest2 for running bcolz tests with Python 2.6!")
-elif sys.version_info[0] == 3:
-    if sys.version_info[1] < 2:
-        exit_with_error("You need Python 3.2 or greater to run bcolz!")
-else:
-    exit_with_error("You need Python 2.6/3.2 or greater to run bcolz!")
-
-# Check if Cython is installed or not (requisite)
-try:
-    import Cython
-    cur_cython_version = Cython.__version__
-    from Cython.Distutils import build_ext
-except:
-    exit_with_error(
-        "You need %(pkgname)s %(pkgver)s or greater to compile bcolz!"
-        % {'pkgname': 'Cython', 'pkgver': min_cython_version})
-
-if cur_cython_version < min_cython_version:
-    exit_with_error(
-        "At least Cython %s is needed so as to generate extensions!"
-        % (min_cython_version))
-else:
-    print("* Found %(pkgname)s %(pkgver)s package installed."
-          % {'pkgname': 'Cython', 'pkgver': cur_cython_version})
+from os.path import join, abspath
+from glob import glob
+import sys
+import re
+
+from setuptools import setup, Extension, find_packages
+from setuptools.command.build_ext import build_ext
+
 
 # Prevent numpy from thinking it is still in its setup process:
 __builtins__.__NUMPY_SETUP__ = False
 
-# Check for NumPy
-check_import('numpy', min_numpy_version)
-
-# Check for Numexpr
-numexpr_here = False
-try:
-    import numexpr
-except ImportError:
-    print_warning(
-        "Numexpr is not installed.  For faster bcolz operation, "
-        "please consider installing it.")
-else:
-    if numexpr.__version__ >= min_numexpr_version:
-        numexpr_here = True
-        print("* Found %(pkgname)s %(pkgver)s package installed."
-              % {'pkgname': 'numexpr', 'pkgver': numexpr.__version__})
-    else:
-        print_warning(
-            "Numexpr %s installed, but version is not >= %s.  "
-            "Disabling support for it." % (
-                numexpr.__version__, min_numexpr_version))
 
-########### End of checks ##########
+class BuildExtNumpyInc(build_ext):
+    def build_extensions(self):
+        from numpy.distutils.misc_util import get_numpy_include_dirs
```

Comment:
While tempting, the above relies on the current Numpy internal structure which is not guaranteed to retain backwards compatibility. The Numpy manual suggests to use get_numpy_include_dirs  ...